src/main/kotlin/com/pubnub/api/subscribe/eventengine/effect/ReceiveReconnectEffect.kt

```diff
@@ -0,0 +1,66 @@
+package com.pubnub.api.subscribe.eventengine.effect
+
+import com.pubnub.api.PubNubException
+import com.pubnub.api.endpoints.remoteaction.RemoteAction
+import com.pubnub.api.eventengine.EventQueue
+import com.pubnub.api.eventengine.ManagedEffect
+import com.pubnub.api.subscribe.eventengine.event.Event
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.ScheduledFuture
+import java.util.concurrent.TimeUnit
+
+class ReceiveReconnectEffect(
+    private val remoteAction: RemoteAction<ReceiveMessagesResult>,
+    private val eventQueue: EventQueue,
+    private val policy: RetryPolicy,
+    private val executorService: ScheduledExecutorService,
+    private val attempts: Int,
+    private val reason: PubNubException?
+) : ManagedEffect {
+
+    @Transient
+    private var scheduled: ScheduledFuture<*>? = null
+
+    @Transient
+    private var cancelled = false
+    override fun runEffect() {
```

Comment:
`@Synchronized` is missing I think in here and in cancel

Reply:
ok. I will add those.